Products/CMFPlone/i18nl10n.py

```diff
@@ -186,34 +193,22 @@ def ulocalized_time(time, long_format=None, time_only=False, context=None,
         return time.strftime(formatstring)
 
     # get the format elements used in the formatstring
-    formatelements = _interp_regex.findall(formatstring)
-
-    # reformat the ${foo} to foo
-    formatelements = [el[2:-1] for el in formatelements]
+    formatelements = {el[2:-1] for el in _interp_regex.findall(formatstring)}
 
     # add used elements to mapping
-    elements = [e for e in formatelements if e in datetime_formatvariables]
-
-    # add weekday name, abbr. weekday name, month name, abbr month name
-    week_included = True
-    month_included = True
-
-    name_elements = [e for e in formatelements if e in name_formatvariables]
-    if not ('a' in name_elements or 'A' in name_elements):
-        week_included = False
-    if not ('b' in name_elements or 'B' in name_elements):
-        month_included = False
-
+    elements = formatelements & datetime_formatvariables
     for key in elements:
         mapping[key] = time.strftime('%' + key)
 
-    if week_included:
+    # add weekday name, abbr. weekday name, month name, abbr month name
+    name_elements = formatelements & name_formatvariables
+    if bool({'a', 'A'} & name_elements):
```

Comment:
Probably `bool` is not needed, but it is fine

Reply:
yes, i due to refactoring I had first the bool and it sticked

Reply:
fixed